src/dev/build/tasks/copy_source_task.ts

```diff
@@ -60,5 +65,55 @@ export const CopySource: Task = {
         'opensearch_dashboards.d.ts',
       ],
     });
+
+    /*
+     * Dependencies linked using `file:` in the `yarn.lock` file are
+     * referenced with relative paths. Moving them to the `build`
+     * folder, these paths need to be transformed.
+     */
+    const writeStream = createWriteStream(join(buildRoot, 'yarn.lock'));
+    const writeLine = (line: string) =>
+      new Promise((resolve) => {
+        if (writeStream.write(line + '\n')) return resolve();
+        writeStream.once('drain', resolve);
+      });
+
+    /*
+     * Dependency patterns are comma separated `name@range` values, followed
+     * by a colon, like `AAAA:` and `AAAA, BBBB, CCCC:`
+     * Some are wrapped in quotes, like `"AAAA":` and `"AAAA, BBBB, CCCC":`
+     *
+     * The linked file will have `@file:` in it and the inner strings will
+     * not have commas or colons.
+     *
+     * Zero-width lookbehind and lookaheads allow getting only the part
+     * required for processing the patterns. `\S` is to avoid matching
+     * spaces after commas.
+     */
+    const linkedFileDepPattern = /(?<=^"?|,\s*)(\S+?[^,:]+)@file:([^,:]+)(?=,\s*|"?:\s*$)/g;
```

Comment:
This comment is great and super helpful. But a test file with a few expected patterns that demonstrates the intended regex behavior would be better. Particularly to help us when we eventually update yarn versions (I'm not sure if the `yarn.lock` patterns are intended to be parse-able as part of yarn's semver'd API.

Reply:
Created a unit test for this.